src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -3328,11 +3387,28 @@ static void updateEntity
   if (strcasecmp(action.c_str(), "replace") == 0)
   {
     // toSet: { A1: { ... }, A2: { ... } }
-    int now = getCurrentTime();
-    updatedEntity.append("$set", BSON(ENT_ATTRS             << toSetObj <<
-                                      ENT_ATTRNAMES         << toPushArr <<
-                                      ENT_MODIFICATION_DATE << now <<
-                                      ENT_LAST_CORRELATOR   << fiwareCorrelator));
+	BSONObjBuilder replaceSet;
+    int            now = getCurrentTime();
+
+    // This avoids strange behavior like as for the location, as reported in the #1142 issue
+    // In order to enable easy append management of fields (e.g. location, dateExpiration),
+    // it could be better to use a BSONObjBuilder instead the BSON stream macro below.
+    if (replaceDate)
+    {
+      updatedEntity.append("$set", BSON(ENT_ATTRS                   << toSetObj <<
+    	                                ENT_ATTRNAMES               << toPushArr <<
+    	                                ENT_MODIFICATION_DATE       << now <<
+										ENT_EXPIRATION              << currentDateExpiration <<
```

Comment:
Github diff shows overindent here. Mabye you are using tabs instead of spaces? Please check it and use the later.

(Also in other cases in this same PR)

Reply:
Fixed in 5a4703d